src/ducks/paymentMethods.duck.js

```diff
@@ -1,2 +1,182 @@
+        stripeCustomer: payload,
+      };
+    case CREATE_STRIPE_CUSTOMER_ERROR:
+      console.error(payload);
+      return {
+        ...state,
+        createStripeCustomerError: payload,
+        createStripeCustomerInProgress: false,
+      };
+    case ADD_PAYMENT_METHOD_REQUEST:
+      return { ...state, addPaymentMethodError: null, addPaymentMethodInProgress: true };
+    case ADD_PAYMENT_METHOD_SUCCESS:
+      return {
+        ...state,
+        addPaymentMethodInProgress: false,
+        stripeCustomer: payload,
+      };
+    case ADD_PAYMENT_METHOD_ERROR:
+      console.error(payload);
+      return {
+        ...state,
+        addPaymentMethodError: payload,
+        addPaymentMethodInProgress: false,
+      };
+
+    case DELETE_PAYMENT_METHOD_REQUEST:
+      return { ...state, deletePaymentMethodError: null, deletePaymentMethodInProgress: true };
+    case DELETE_PAYMENT_METHOD_SUCCESS:
+      return {
+        ...state,
+        deletePaymentMethodInProgress: false,
+      };
+    case DELETE_PAYMENT_METHOD_ERROR:
+      console.error(payload);
+      return {
+        ...state,
+        deletePaymentMethodError: payload,
+        deletePaymentMethodInProgress: false,
+      };
+    default:
+      return state;
+  }
+}
+
+// ================ Action creators ================ //
+
+export const stripeCustomerCreateRequest = () => ({ type: CREATE_STRIPE_CUSTOMER_REQUEST });
+
+export const stripeCustomerCreateSuccess = stripeCustomer => ({
+  type: CREATE_STRIPE_CUSTOMER_SUCCESS,
+  payload: stripeCustomer,
+});
+
+export const stripeCustomerCreateError = e => ({
+  type: CREATE_STRIPE_CUSTOMER_ERROR,
+  payload: e,
+  error: true,
+});
+
+export const addPaymentMethodRequest = () => ({ type: ADD_PAYMENT_METHOD_REQUEST });
+
+export const addPaymentMethodSuccess = stripeCustomer => ({
+  type: ADD_PAYMENT_METHOD_SUCCESS,
+  payload: stripeCustomer,
+});
+
+export const addPaymentMethodError = e => ({
+  type: ADD_PAYMENT_METHOD_ERROR,
+  payload: e,
+  error: true,
+});
+
+export const deletePaymentMethodRequest = () => ({ type: DELETE_PAYMENT_METHOD_REQUEST });
+
+export const deletePaymentMethodSuccess = stripeCustomer => ({
+  type: DELETE_PAYMENT_METHOD_SUCCESS,
+  payload: stripeCustomer,
+});
+
+export const deletePaymentMethodError = e => ({
+  type: DELETE_PAYMENT_METHOD_ERROR,
+  payload: e,
+  error: true,
+});
+
+// ================ Thunks ================ //
+
+export const createStripeCustomer = params => (dispatch, getState, sdk) => {
+  const stripePaymentMethodId = params.setupIntent.payment_method;
+  dispatch(stripeCustomerCreateRequest());
+  return sdk.stripeCustomer
+    .create({ stripePaymentMethodId }, { expand: true, include: ['defaultPaymentMethod'] })
+    .then(response => {
+      const stripeCustomer = response.data.data;
+      dispatch(stripeCustomerCreateSuccess(stripeCustomer));
+      return stripeCustomer;
+    })
+    .catch(e => {
+      log.error(e, 'create-stripe-user-failed');
```

Comment:
Don't log plain error. It can contain internal reference loops (Axios related loops, if I remember correctly). Those can be quite problematic if data needs to be serialized with `JSON.stringify`.

=> use `storableError(e)`